ecosystem/sep-0039.md

```diff
@@ -0,0 +1,182 @@
+
+A standard storage mechanism will help these marketplaces and wallets render or otherwise represent the NFT within their applications, enhancing their transferability.
+
+
+## Abstract
+We present an efficient encoding mechanism to store arbitrary binary data on the Stellar ledger via the `ManageData` operation, specifically catering towards NFTs. It will concatenate a special SEP-39 header alongside a [Media Type](https://www.iana.org/assignments/media-types/media-types.xhtml) containing asset metadata alongside the aforementioned encoded-text as keys and raw binary data as values.
+
+
+## Specification
+First, we will discuss a common NFT issuance model and its ownership implications. This is not a required model, but sets the stage for why a standardized encoding of digital assets is necessary.
+
+**Warning**: Because of how indexing works (that is, each data entry for the asset begins with an index), an account can only hold a single SEP-39 asset at a time. Users should also take care not to have colliding keys that may interfere with ordering, and should lean on the (optional) checksum parameter if they're worried about decoding inconsistencies.
+
+### NFT Issuance
+Given an arbitrary digital asset, here's how issuance typically works:
+
+  1. An account is created to represent the digital asset on the ledger
+  2. The account stores the entire digital asset within its managed data entries.
+  3. The account issues a Stellar asset representing ownership over the digital asset we just stored.
+  4. Any account holding that Stellar asset lays claim to the digital asset.
+
+Notice that ownership of the token directly conveys ownership of a unit of the asset, entirely within the Stellar ledger. Other scenarios such as locking down the issuing account to avoid supply increases, ownership transfers via payments, etc. are built on top of this basic model and are thus extraneous to this SEP.
+
+Note also that this SEP does not address concerns around *immutability*. While there is an optional checksum component (addressed later), the account itself is free to modify the data at-will if it is not locked.
+
+Our exclusive focus in this SEP is Step 2: **storing the asset within the account.**
+
+### Data Format
+We begin by describing the raw data format, then dive into each component.
+
+    00|version|metadata length|media-type;param1=value;param2=value;etc<binary data>
+
+Here, `|` is not a character, but rather a semantic divider used to separate the components. The following sections break down each component in detail.
+
+#### Index
```

Comment:
I'm confused by this. Are you suggesting printing out the index raw via `000`, `001` etc? This seems inefficient. In my estimation this should definitely be converted to using two bytes. The way I've been doing it via JS is 
```
const index = Buffer.alloc(2)
index.writeIntBE({index}, 0, 2)
```

Then later on to actually sort the keys you can do
```
parseInt(key.slice(0, 2).toString('hex'), 16)
```

The method of using three chars burns one extra char per manage data entry.

Was the thinking that by using raw string indexes you could avoid collision with other non sep39 key value entries?